### PR TITLE
[ESIMD][NFC] Rework the L1/L2 cache hints passing across internal funcs

### PR DESCRIPTION
**Pull request by v-klochkov**
This is the 1st patch in the upcoming series of similar patches.

Cache-hints:
* cache_hint and cache_level were also moved to memory_properties.hpp
* added check_cache_hints() function accepting PropertyListT instead of
  L1/L2 template parameters.

Restructures in block_load:
* block_load_impl() functions now accept template param PropertyListT
  instead of L1H, L2H and alignment FlagsT.
* only block_load_impl functions call check_cache_hints() now.
* Replaced the uses of lsc.load.stateless with lsc.load.merge.stateless
  It does not change the GPU code-gen, it is identical.

Restructures in block_store:
* block_store_impl() functions now accept template param PropertyListT
  instead of L1H, L2H and alignment FlagsT.

### CODE REVIEW

**Comment by sarnex** [comment]
The Linux failures seem like they might be related. If not let me know and I will review this now.

**Comment by v-klochkov** [comment]
> The Linux failures seem like they might be related. If not let me know and I will review this now.

It is interesting that SYCLOS has extra new GPU driver now - 1.3.28454.
I used agama 1.3.28202 and the test passed locally. The GPU asm is identical with and without my patch.
The test also passed with non-public new driver - 1.3.28597

File: sycl/test/esimd/memory_properties.cpp [hunk 1]
```diff
@@ -104,26 +104,26 @@ test_block_load(AccType &acc, LocalAccType &local_acc, float *ptrf,
...
+  // CHECK: call <4 x i32> @llvm.genx.lsc.load.merge.stateless.v4i32.v1i1.v1i64(<1 x i1> {{[^)]+}}, i8 0, i8 5, i8 2, i16 1, i32 0, i8 3, i8 4, i8 2, i8 0, <1 x i64> {{[^)]+}}, i32 0, <4 x i32> {{[^)]+}})
   auto d2 = block_load<int, N>(ptri, byte_offset32, props_a);
 
-  // CHECK: call <4 x float> @llvm.genx.lsc.load.stateless.v4f32.v1i1.v1i64(<1 x i1> {{[^)]+}}, i8 0, i8 2, i8 1, i16 1, i32 0, i8 3, i8 4, i8 2, i8 0, <1 x i64> {{[^)]+}}, i32 0)
+  // CHECK: call <4 x float> @llvm.genx.lsc.load.merge.stateless.v4f32.v1i1.v1i64(<1 x i1> {{[^)]+}}, i8 0, i8 2, i8 1, i16 1, i32 0, i8 3, i8 4, i8 2, i8 0, <1 x i64> {{[^)]+}}, i32 0, <4 x float> {{[^)]+}})
   auto d3 = block_load<float, N>(ptrf, byte_offset64, props_b);
 
-  // CHECK: call <4 x float> @llvm.genx.lsc.load.stateless.v4f32.v1i1.v1i64(<1 x i1> {{[^)]+}}, i8 0, i8 5, i8 2, i16 1, i32 0, i8 3, i8 4, i8 2, i8 0, <1 x i64> {{[^)]+}}, i32 0)
+  // CHECK: call <4 x float> @llvm.genx.lsc.load.merge.stateless.v4f32.v1i1.v1i64(<1 x i1> {{[^)]+}}, i8 0, i8 5, i8 2, i16 1, i32 0, i8 3, i8 4, i8 2, i8 0, <1 x i64> {{[^)]+}}, i32 0, <4 x float> {{[^)]+}})
```
Comment (sarnex) [inline sycl/test/esimd/memory_properties.cpp:116]: do the normal lsc load and the passthrough lsc load have the exact same requirements and supported cases? if so, this seems like a good simplication to me

**Review by sarnex — COMMENTED** [review]
looks great, really cool cleanup. just a minor question

File: sycl/test/esimd/memory_properties.cpp (hunk 1 above)
Comment (v-klochkov) [reply sycl/test/esimd/memory_properties.cpp:116]: Yes, exactly the same, and lowered to the same visa instruction. I tried some tests to check that too.
Later I would replace the use of lsc.load.stateless with lsc.load.merge.stateless for `gather` API too.

**Review by sarnex — APPROVED** [review]

**Pull request merged**